### PR TITLE
Arm64

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds ARM64 support

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
Note that there are few relevant checkpatch warnings (among non-relevant ones).

**Comment by jenswi-linaro** [comment]
Review comments addressed, except the open question in core/arch/arm/kernel/thread_private.h.

File: core/arch/arm/include/arm64_macros.S [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		.endif
+	.endm
+	.macro __do_reg op1, op2, reg_bytes, base_reg, base_offs, reg
+		.if \reg_bytes == 8
+			\op1\op2	x\reg, [\base_reg, #\base_offs]
+		.else
+			\op1\op2	w\reg, [\base_reg, #\base_offs]
+		.endif
+	.endm
+	.macro _do_regs op, reg_bytes, base_reg, base_offs, from_regnum, \
```
Comment (jbech-linaro) [inline core/arch/arm/include/arm64_macros.S]: op is coming the instruction (st/ld) and in the subsequent calls it's passed/labelled as op1. I think it's a bit hard to follow this. Could we simply rename op and op1 to "ins" or something else that differentiate it a bit and gives a hint that it's a "prefix" of an assembler instruction?

Comment (jbech-linaro) [inline core/arch/arm/include/arm64_macros.S]: Also, those macro's are quite big and also uses some recursion. I think a comment in clear text saying what is going on would make things a bit easier when trying to follow the macro(s).

**Comment by jenswi-linaro** [comment]
Review comments addressed, except the open question in core/arch/arm/kernel/thread_private.h.

**Comment by jenswi-linaro** [comment]
Review comments addressed.

**Comment by pascal-brand-st-dev** [comment]
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jenswi-linaro** [comment]
Rebased and tags applied.

**Comment by jenswi-linaro** [comment]
Rebased. Baseline seems broken on qemu virt, these changes doesn't add any new errors.
